File: app/src/main/java/io/github/skywalkerdarren/simpleaccounting/model/dao/TypeDao.java
package io.github.skywalkerdarren.simpleaccounting.model.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;
import java.util.UUID;

import io.github.skywalkerdarren.simpleaccounting.model.entity.Bill;
import io.github.skywalkerdarren.simpleaccounting.model.entity.Type;

@Dao
public interface TypeDao {
    @Query("SELECT * FROM type WHERE uuid == :uuid")
    Type getType(UUID uuid);
    @Query("SELECT * FROM type WHERE is_expense == :isExpense")
    List<Type> getTypes(Boolean isExpense);
    @Query("DELETE FROM type WHERE uuid = :uuid")
    void delType(UUID uuid);
    @Insert
    void newType(Type type);
}

File: app/src/main/java/io/github/skywalkerdarren/simpleaccounting/model/dao/TypeDao.kt
package io.github.skywalkerdarren.simpleaccounting.model.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import io.github.skywalkerdarren.simpleaccounting.model.entity.Type
import java.util.*

@Dao
interface TypeDao {
    @Query("SELECT * FROM type WHERE uuid == :uuid")
    fun getType(uuid: UUID): Type?

    @Query("SELECT * FROM type WHERE is_expense == :isExpense")
    fun getTypes(isExpense: Boolean): List<Type>?

    @Query("DELETE FROM type WHERE uuid = :uuid")
    fun delType(uuid: UUID)

    @Insert
    fun newType(type: Type)
}